authority_test.go

```diff
@@ -0,0 +1,130 @@
+package tlsconfig
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"code.cloudfoundry.org/tlsconfig/certtest"
+)
+
+func TestEmptyPool(t *testing.T) {
+	t.Parallel()
+	pool, err := FromEmptyPool().Build()
+	if err != nil {
+		t.Fatalf("unexpected error when building empty pool: %q", err)
+	}
+
+	if size := len(pool.Subjects()); size != 0 {
+		t.Errorf("expected pool to be empty but it had %d certificates", size)
+	}
+}
+
+func TestSystemPool(t *testing.T) {
+	t.Parallel()
+	pool, err := FromSystemPool().Build()
+	if err != nil {
+		t.Fatalf("unexpected error when building system pool: %q", err)
+	}
+
+	if size := len(pool.Subjects()); size == 0 {
+		t.Error("expected pool to contain something but it did not")
+	}
+}
+
+func TestWithCert(t *testing.T) {
+	t.Parallel()
+
+	auth, err := certtest.BuildCA("theauthority")
+	if err != nil {
+		t.Fatalf("failed to create CA: %s", err)
+	}
+
+	cert, err := auth.Certificate()
+	if err != nil {
+		t.Fatalf("failed to create certificate: %s", err)
+	}
+
+	pool, err := FromEmptyPool(
+		WithCert(cert),
+	).Build()
+	if err != nil {
+		t.Fatalf("unexpected error when building pool: %q", err)
+	}
+
+	if size := len(pool.Subjects()); size != 1 {
+		t.Errorf("expected pool to have size 2 but it had %d certificates", size)
```

Comment:
```suggestion
		t.Errorf("expected pool to have size 1 but it had %d certificates", size)
```

Reply:
Thanks! I introduced `have` and `want` variables to avoid this kind of mistake in the future.